openage/convert/dataformat/data_definition.py

```diff
@@ -124,6 +124,114 @@ def generate_csv(self, genfile):
             reprtxt="csv for %s" % self.name_struct,
         )]
 
+    def generate_sprite(self, genfile):
+        """
+        create a text snippet to represent the sprite data
+        """
+
+        # TODO: sprite files also need to contain metadata extracted from
+        #       empires.dat. Now, it only contains frame geometry information
+        #       coming from slp files. Some stuff is hardcoded, for now.
+
+        # pylint: disable=too-many-locals
+
+        # the resulting file content
+        txt = []
+
+        # create the information comment header
+        txt.extend([
+            "# File version", "\n",
+            "version 0 ", "\n",
+            "\n",
+            ])
+
+        # image file reference, relative to this file
+        txt.extend([
+            "# Image file path", "\n",
+            "image 0 ", "./", self.name_data_file, ".png", "\n",  # TODO: this sucks
```

Comment:
multiple images can be referenced. For example, buildings have separate source SLPs for building sprite and building shadow. I think we need a for-loop here.

Reply:
Yes, but this information only comes from the `.dat` file. Same goes for layers. I don't think there's a quick way to hack this in without actually dealing with parsing and distributing the data from the `.dat` file. Or am I missing something?

Reply:
True, but that should not matter here. `DataDefinition` should only print data it has received anyway. If it has not received any layer members, it doesn't create a line in `txt`. You can assume that the converter did extract and ordered the data before.

Reply:
@brisvag the assumption for this format is that we have access to all information, even from the `dat` file. because when we do the conversion, we do have all necessary links between frames, drsfiles, units etc.